tests/util/helper/cmd/main.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	"io"
+
+	"istio.io/istio/mixer/cmd/shared"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/tests/util/registry"
+)
+
+type localRegistrySetupArgs struct {
+	localRegistryPort uint16
+	kubeconfig        string
+}
+
+// closerRegister is a non thread safe struct to register io.Closers
+type closerRegister struct {
+	closers []io.Closer
+}
+
+func (cr *closerRegister) register(closer io.Closer) {
+	cr.closers = append(cr.closers, closer)
+}
+
+func (cr *closerRegister) stop() {
+	if len(cr.closers) == 0 {
+		return
+	}
+	fmt.Println("Hit CTRL-C to interrupt")
+	stop := make(chan os.Signal, 1)
```

Comment:
Suggestion: 

1) Move the logic to wait for SIGTERM to main
2) Rename this method to `Close() error` (i.e. io.Closer) and make it just close all of the closers.
3) In main, after receiving SIGTERM, just call `cr.close()`

Reply:
I'm confused ... my suggestion should not change behavior in any way.  If it hangs, it should hang in both cases.